controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -391,6 +394,16 @@ public ActivateResult deploy(JobId job, boolean deploySourceVersions) {
             // Record the quota usage for this application
             var quotaUsage = deploymentQuotaUsage(zone, job.application());
 
+            // For direct deployments use the full application ID, but otherwise use just the tenant and application as
+            // the source since it's the same application, so it should have the same warnings
```

Comment:
Unless of course the user uses some deprecated feature that is only enabled in certain environment/region, but it's much harder to aggregate all that without also duplicating the warnings that are present in all/several deployments. :see_no_evil:

Reply:
~(˘▾˘)~